Drilldown method to show any menu

SassNinja opened the pull request:
This PR adds the new method `_showMenu` to the drilldown component for opening any (sub)menu in it **without** clicking through every menu on the way to it.

I usually like the drilldown menu because it's a nice and intuitive navi which I often use in the off-canvas. But it has a huge downside: there's currently no method to jump into a specific submenu without clicking through several submenus.

I've added a visual test in this PR that shows the difference between the current `_show` and the new `_showMenu`. You'll see what I mean when you check the 'Use current show' checkbox and click the buttons C, B, X (in this order) and then use the back links.
http://localhost:3000/drilldown/drilldown-show-menu.html

The new method is probably a great feature for everyone who needs to e.g. auto open the current page in the drilldown if using it as page navi.

@IamManchanda @kball @brettsmason what do you think?

Commit:
Added check in orbit.js to make sure slides aren't animating before hiding them when resizing the window. This should fix #9788

Commit:
Update foundation.orbit.js

SassNinja committed:
add nested off-canvas feature + docs and visual test

SassNinja committed:
fix nested overlay transition so it doesn't jump on transition end

SassNinja committed:
Add nested detection and fixed position emulation for nested element

SassNinja committed:
Create inner side shadow (incl. mixin) for nested push off-canvas element

SassNinja committed:
Fix off-canvas resize event listener

The height should not only be reset if the element is open but every time so the value is correct on opening.

SassNinja committed:
Fix nested off-canvas position issue in IE

Since position fixed doesn't work for nested off-canvas elements, it has to be position absolute. Surprisingly (not) the IE doesn't show an opened off-canvas element unless absolute is explicitly set.

SassNinja committed:
Make in-canvas position override important

Due to the IE fix where position absolute gets explicitly set, in-canvas has to use !important to override this reliably.

SassNinja committed:
use first() on siblings result

Since it doesn't make sense to have more than one match for the content element I'm using first() to make this sure.

SassNinja committed:
Fix emulateFixedPosition() so top & bottom position work correctly

Top position mustn't get any inline height (preserve CSS height) but only an updated top value.
Bottom position is identical to top position but needs another calculation to pseudo fix it to bottom.

SassNinja committed:
Replace off-canvas shadow for push panel with inner box shadow

Using a pseudo ::after element for the off-canvas push panel shadow causes issues if the element's content height is greater than the element's height.
Therefore I've added two additional variables for the push panel and use it to create an inner box shadow.

SassNinja committed:
Adjust in-canvas selector to get rid of some !important

SassNinja committed:
Fix transitionend listener to avoid an issue if prematurely closed

SassNinja committed:
Add a nested option to improve the content ID usage + adjust docs

As it's very hard to determine whether the element is nested when using the content ID option, it's required to set the nested option in this a case.
Therefore the emulateFixePosition function doesn't interfere with a actual position:fixed

SassNinja committed:
Create relative scope detection & use the relative parent to adjust the calculation

If a nested off-canvas element is placed in a container with position:relative (mostly push/pull classes) this must be considered in the calculation of the fixed position emulation.
To detect the relative scope I'm checking if the closest positioned element (offsetParent) has a left css value (typical for push/pull) or is affecting the top coord.
The usage of adjustToLocalScope() is limited to nested elements because non-nested elements doesn't have that scope issue (due to position:fixed)

SassNinja committed:
Prevent FOUC by adding is-closed class in markup

Odaeus committed:
Fix internal link on grid page

kball committed:
Merge pull request #10144 from Odaeus/Odaeus-patch-1

[Docs] Fix internal link on grid page

SassNinja committed:
Merge remote-tracking branch 'stream/develop' into develop

davidluhr committed:
Add else statement inside of ifequal example

It is unclear that else statements are possible inside of {{#ifequal}} helpers to show HTML if the two values are not equal. Because there are {{#ifpage}} and {{#unlesspage}} as separate equal/not equal helpers, users may be confused that there is no equivalent "unlessequal" helper or "!" syntax.

I have successfully used the {{else}} Handlebars helper inside if the {{#ifequal}} Paini helper to achieve the desired "not equal" condition, but only thought to try this as a guess after a lot of unsuccessful searching through documentation on both Handlebars and Panini.

SassNinja committed:
Force overlap transition for nested off-canvas elements to avoid transform scope issues

Using position fixed within a transform scope doesn't work and fixedPositionEmulation + position calculations don't work reliably yet.
Therefore I'm forcing overlap transition when nesting (at the moment).

SassNinja committed:
Add $wrapperInner (prior v6.3) detection + a CSS fix

Regarding the off-canvas docs setup, a detection of the prior off-canvas wrapper inner is necessary.
Besides I'm fixing some specificity issue caused by _page.scss (foundation-docs)

SassNinja committed:
Undo the wrapper inner fix

Since wrapper inner is deprecated and erroneously implemented in the docs, I'm undoing my fix.
Once the docs got fixed, the push transition is going to work as expected.

SassNinja committed:
Remove deprecated wrapper inner from docs layout

Firstly the implementation was erroneous (content mustn't be a sibling of the wrapper but a child) and secondly the wrapper inner is deprecated.
Therefore I'm removing it what makes the push transitions on the off-canvas docs page working again as expected.

gnuletik committed:
Create media-queries.md

SassNinja committed:
Remove content classes on close, fix transitionend listener & set default transform none for content

Now the content container has none transform by default so position fixed works as soon as an push off-canvas is closed.
Besides the transitionend listener is now fixed so it isn't bound to only the content or the element but to both of them.
The content classes specification is now part of the setup (this.contentClasses) for an improved readability.

SassNinja committed:
Restore transform scope for push transitions so transitionend works correctly

Due to the content's initial transform:none the transitionend listener didn't work for the content container.
This is fixed now by using the has-transition-push class that is also used in the positions mixin to limit the translate to push transitions.

SassNinja committed:
Remove the transitionend listener for the content container

Since the limited version doesn't enable push transitions for nested elements, there's no need to listen on the content because the element always gets moved.

derweili committed:
paramenter description fixed

IamManchanda committed:
Closes #10189, Add a dependency alert of motion ui for orbit!

kball committed:
Merge pull request #10185 from GNUletik/patch-1

Add Foundation.MediaQuery.is to the docs

kball committed:
Merge pull request #10195 from derweili/patch-1

paramenter description fixed

kball committed:
Merge pull request #10197 from zurb/docs/orbit-dependency-motionui

Closes #10189, Add a dependency alert of motion ui for orbit!

kball committed:
Merge pull request #10156 from davidluhr/patch-1

Add else statement inside of ifequal example

kball committed:
Add css for nested variation of frame

SassNinja committed:
Consider element & content are nested in another content container

brettsmason committed:
Start of reworking flexbox utilities as its own page.

brettsmason committed:
Remove duplicate source order classes from legacy flex grid. These should be just a flex utility.

kball committed:
Add testcase and add qualifiers on basis overrides

kball committed:
Add comment about the hackery and backstory

SassNinja committed:
Merge remote-tracking branch 'upstream/develop' into develop

SassNinja committed:
Get rid of all in-canvas !important

Since the push support for nested elements got reverted the height doesn't have to fight inline CSS anymore. Box shadow and transform override work with a increased CSS specificity now.

IamManchanda committed:
Make source ordering flexible in sass

Added a variable for the count!

IamManchanda committed:
Move Source ordering from flex grid to flexbox utilities page!

IamManchanda committed:
Update docs example as per xy grid classes!

IamManchanda committed:
Update Align self with new example!

SassNinja committed:
Hide close-button for in-canvas and reveal-for-x via CSS

Since reveal-for-x uses now CSS for hiding the related JS for the inline style is removed.

SassNinja committed:
Fix nested reveal and its class adding/removing in JS + replace Foundation global with imported transitionend

IamManchanda committed:
Update visual test with xy grid classes part 1

IamManchanda committed:
Update visual test with xy grid classes part 2!

IamManchanda committed:
Update visual test with xy grid classes part 3!

IamManchanda committed:
Deleted old menu/responsive menu files

Reason
- They are just not working properly
- There are new files in the tests which are working properly!

IamManchanda committed:
Update visual test with xy grid classes part 4!

IamManchanda committed:
Update visual test with xy grid classes part 5!

kball committed:
Merge pull request #10233 from zurb/visual/grid-styling-components

Update visual test with xy grid classes

kball committed:
Now that we swapped global default, foundation-flaot needs to set -flexbox: false

IamManchanda committed:
Code Indentation and end of line!

IamManchanda committed:
Merge pull request #10234 from zurb/fix-foundation-float

Fix foundation-float.css

kball committed:
Merge pull request #10003 from SassNinja/develop

Nested Off-Canvas II

kball committed:
Merge pull request #10209 from zurb/nested-frame-css

Add css for nested variation of frame

IamManchanda committed:
Fix #10229 - Add align item center to `.menu.simple`!

IamManchanda committed:
Move vanilla flex helpers from flex-grid to flex utilities

- Also Add a breakpoint variable to let user disable it if they want!

brettsmason committed:
Merge pull request #10222 from zurb/xy-grid-auto-overgeneralize

Add testcase and add qualifiers on basis overrides

IamManchanda committed:
Update flexbox helpers and source ordering docs!

IamManchanda committed:
Remove codepen links as just not the right codepen for xy grid!

kball committed:
Merge pull request #10224 from zurb/flexbox-utilities-docs

Flexbox utilities docs & Some Sass Changes!

brettsmason committed:
Add docs explaining how to use the xy grid mixins

kball committed:
Merge pull request #10235 from zurb/hotfix/menu-text-issue

Fix #10229 - Add align item center to `.menu.simple`!

kball committed:
Merge pull request #10236 from zurb/xy-grid-semantic-docs

Add docs explaining how to use the xy grid mixins

kball committed:
Handle both spellings for closable/closeable

IamManchanda committed:
Add visual tests for close button!

IamManchanda committed:
Merge pull request #10237 from zurb/fix-closable-bug

Handle both spellings for closable/closeable

kball committed:
Package up 6.4.0-rc5

kball committed:
Fix leftover console logs

kball committed:
One more pass at packaging up rc5

brettsmason committed:
Remove `rem-calc` use for `xy-grid-container`

This allows for more flexible widths, eg `vw`. See https://github.com/zurb/foundation-sites/issues/10141#issuecomment-310682672 for details.

kball committed:
Add viewport meta to new menu visual tests

slipets-a committed:
Changes the Abide pattern to support Mastercard 2-digits BIN

IamManchanda committed:
Add missing viewport tag on all test cases!

IamManchanda committed:
Merge pull request #10245 from zurb/visual-tests/add-viewport

Add missing viewport tag on all test cases!

IamManchanda committed:
Sass Docs: Correct Menu Simple bullet point issue

Thanks Leah Callahan for reporting at QA!

IamManchanda committed:
Merge pull request #10246 from zurb/sass-docs/menu-simple-mixin

Sass Docs: Correct Menu Simple bullet point issue

IamManchanda committed:
Add backwards compatibility for menu-state-active

Keep `active` class but with option to disable the same!

IamManchanda committed:
Add Xy grid classes to nested off canvas!

IamManchanda committed:
Merge pull request #10250 from zurb/tests/add-xy-grid-class-nested-offcanvas

Add Xy grid classes to nested off canvas!

IamManchanda committed:
Sticky Offcanvas: Remove unused `row` and `column` classes!

yannick-milanetto committed:
Fix duplicate CSS property on slider-handle mixin

rafibomb committed:
Merge pull request #10253 from yannick-milanetto/fix/duplicate-css-property-slider

Fix duplicate CSS property on slider-handle mixin

kball committed:
Fix plugin naming for ie9

kball committed:
Fix polyfill for matchMedia

kball committed:
Fix parenthesis placement

IamManchanda committed:
Remove mistakenly added `+` sign!

IamManchanda committed:
Merge pull request #10256 from zurb/docs/print-style-alert

Remove mistakenly added `+` sign!

kball committed:
Brute force the auto override issue

IamManchanda committed:
Fixes the wrong declaration of parameters for default!

IamManchanda committed:
Merge github.com:zurb/foundation-sites into docs/custom-block-grid

kball committed:
Fix destroy reference for IE as well

IamManchanda committed:
Merge pull request #10258 from zurb/docs/custom-block-grid

Fixes the wrong declaration of parameters for default!

IamManchanda committed:
Update offcanvas transitions copy!

kball committed:
reset menu input to be inline-block

IamManchanda committed:
Merge pull request #10260 from zurb/docs/off-canvas-transitions

Update offcanvas transitions copy!

IamManchanda committed:
Merge pull request #10255 from zurb/responsive-menu-not-working-64

Responsive menu not working 64

kball committed:
Merge pull request #10239 from zurb/brettsmason-xy-grid-container

Remove `rem-calc` use for `xy-grid-container`

IamManchanda committed:
Merge pull request #10259 from zurb/issue-10248-menu-alignment

reset menu input to be inline-block

kball committed:
Add  class to ease padding containers below max-width

IamManchanda committed:
Add menu centered for backward compatibility!

IamManchanda committed:
Update test case copy for `active`!

IamManchanda committed:
Remove `display: flex`!

kball committed:
Merge pull request #10247 from zurb/back-compat/menu-state-active

Add backwards compatibility for menu-state-active & menu-centered

IamManchanda committed:
Merge pull request #10257 from zurb/fix-auto-overrides

Brute force the auto override issue

IamManchanda committed:
Update the test case with better grid styling
Also added a alert to resize window to know difference!

kball committed:
Merge pull request #10261 from zurb/padding-on-grid-containers

Add  class to ease padding containers below max-width

kball committed:
Update customizer config

Odaeus committed:
Remove background color from styled <legend>

Fixes #10254.

Removing the colour makes no difference to the display and allows the element to be placed on areas with different background colours without having to override the element's CSS.

yannick-milanetto committed:
Merge pull request #1 from zurb/develop

Update forked repo with last changes from original repo

yannick-milanetto committed:
Fix Interchange when replacing background image containing parenthesis

kball committed:
Merge branch 'develop' into release/6.4.0

kball committed:
Merge branch 'master' into develop

kball committed:
Merge branch 'master' into release/6.4.0

kball committed:
Package up 6.4.0

kball committed:
Update customizer index file

IamManchanda committed:
Add codepen for horizontal alignment!

IamManchanda committed:
Add codepens for vertical alignment and central alignment!

IamManchanda committed:
Add codepen for central alignment!

IamManchanda committed:
Rearrange the ordering of the content in a correct way!

I messed this in Previous PR but now corrected it!

kball committed:
Merge branch 'master' into develop

tdhartwick committed:
footer updates

IamManchanda committed:
Add codepen for vanilla helpers, responsive classes & source ordering!

Commit:
updates styles and fixes links

Commit:
Merge branch 'develop' into footer-update

Commit:
Merge branch 'footer-update' into develop

IamManchanda committed:
v6.4: Quick codepen update for issue template

IamManchanda committed:
Add RTL pens!

Commit:
adds description to form

IamManchanda committed:
Merge pull request #10269 from zurb/issue-template/add-64-codepen

v6.4: Quick codepen update for issue template

rafibomb committed:
Merge pull request #10262 from Odaeus/odaeus-legend-color

Remove background color from styled <legend>

kball committed:
Fix issue #10279 by removing mistaken '&' for top level selector

kball committed:
Fix customizer js

kball committed:
Workaround issue leading to customizer js fail

IamManchanda committed:
Nest accordion arrows block within `.accordion-menu`!

kball committed:
Merge pull request #10288 from zurb/menu/left-right-arrows

Nest accordion arrows block within `.accordion-menu`!

kball committed:
Merge pull request #10286 from zurb/IS10279-fix-scss-for-ruby-sass

Fix issue #10279 by removing mistaken '&' for top level selector

kball committed:
Merge pull request #10264 from yannick-milanetto/fix/url_with_parenthesis_interchange

Fix Interchange when used on images path containing parenthesis

kball committed:
Merge pull request #10241 from slepets-a/change-abide-card-pattern-for-new-2-digits-bins

Changes the Abide pattern to support Mastercard 2-digits BIN

kball committed:
Tweak ad js to have default topic

kball committed:
Tweak ad js to have default topic

kball committed:
Update docs version to re-introduce TOC ad unitg

kball committed:
Merge pull request #10268 from zurb/codepen/flexbox-utilities

Add Codepen's for flexbox utilities

Commit:
updates responsicve footer styles

Commit:
Merge branch 'master' of https://github.com/zurb/foundation-sites

Commit:
updates latest update on homepage and fixes nave hover color

kball committed:
Merge branch 'master' into develop

kball committed:
Fix global media query reference

kball committed:
Sticky visual tests should use float build for now

kball committed:
Merge pull request #10292 from zurb/fix-global-reference-in-foundation.core

Fix global media query reference

IamManchanda committed:
Make Grid Page Styling also like flexbox utilities!

IamManchanda committed:
Update comment!

wulfsolter committed:
Custom Pattern and Validator code fix

JS example fixed - set defaults **before** initializing foundation. Won't work otherwise.

kball committed:
Remove unneeded declaration in docs

kball committed:
Update sass docs page

kball committed:
Merge pull request #10296 from zurb/redesign/xy-grid-page

Make Grid Page Styling also like flexbox utilities!

kball committed:
Merge pull request #10307 from zurb/fix-xy-grid-classes-docs

Fix xy grid classes docs

kball committed:
Merge pull request #10298 from wulfsolter/patch-1

Custom Pattern and Validator code fix

kball committed:
Quick fix for js docs

kball committed:
Merge pull request #10310 from zurb/quick-fix-js-docs

Quick fix for js docs

kball committed:
Merge branch 'master' into develop

kball committed:
Fix comments to allow octophant to find some vars that were being missed in settings generation

kball committed:
Merge pull request #10311 from zurb/fix-octophant-finding-settings-vars

Fix comments to allow octophant to find some vars that were being mis…

kball committed:
Package up 6.4.1 release

IamManchanda committed:
Update codepen with patch update version - 6.4.1

IamManchanda committed:
Merge pull request #10314 from zurb/codepen/6.4.1

Update codepen with patch update version - 6.4.1

codetheorist committed:
Update gulp-postcss version to 7.0.0 - Issue 10322

IamManchanda committed:
Merge pull request #10325 from codetheorist/update-gulp-postcss

Update gulp-postcss version to 7.0.0 - Issue 10322

codetheorist committed:
Update 'Putting it all together' link in off-canvas docs - Issue 10327

codetheorist committed:
Update menu docs to specify adding '.icons' class as well as layout-specific class.

codetheorist committed:
Update wording of docs

codetheorist committed:
Update codepen links in menu icons docs

IamManchanda committed:
Update codepen links with multiple codepen's!

IamManchanda committed:
Add backward for using icons without `.icons` class!

IamManchanda committed:
Add test for backward compatible feature!

kball committed:
Merge pull request #10331 from codetheorist/update-menu-docs

Update menu docs/examples and Add backward compatibility for menu icons

IamManchanda committed:
Update docs on push/pull - source ordering!

kball committed:
Merge pull request #10332 from zurb/docs/push-pull-bye-bye

Update docs on push/pull - source ordering!

kball committed:
Merge pull request #10329 from codetheorist/off-canvas-docs

Update 'Putting it all together' link in off-canvas docs - Issue 10327

Commit:
Fixed #10333

IamManchanda committed:
Merge pull request #10335 from hikouki/develop

Fixed #10333

LeoColomb committed:
remove `flex-grid-column()` duplicate `max-width`

Very small fix inside `flex-grid-column()` mixin: `max-width` fix is added twice: call to `flex-grid-size()` already import this fix once.

IamManchanda committed:
Merge pull request #10341 from LeoColomb/patch-2

Remove `flex-grid-column()` duplicate `max-width`

Commit:
Fixed #10344

Reset `#{$global-left}`.

kball committed:
Merge pull request #10345 from hikouki/develop

Fixed #10344

IamManchanda committed:
Move typescale utility from prototype to typography
Still keeping it disabled by default
Fix #10301

IamManchanda committed:
Remove extend and add typescale by default!

IamManchanda committed:
Move docs from prototype to helpers!

kball committed:
Merge pull request #10351 from zurb/hotfix/add-typescale

Hotfix: Move typescale from prototype to typography and enable default!

kball committed:
Add training section to readme

kball committed:
Merge pull request #9881 from dwt47/develop

Fix bug on Orbit resize during transition

Commit:
Fixed #10284

Before calculating the boundary values of the two handlers,
it was necessary to calculate the vertical position.

Commit:
Add docs for responsive breakpoints

IamManchanda committed:
Clean up the change added in `_settings` by mistake!

IamManchanda committed:
Update docs copy on prefixing responsive classes from single to all helpers!

kball committed:
Merge pull request #10355 from hikouki/10284

#10284 Fixed bug of vertical two handlers slider.

IamManchanda committed:
Merge pull request #10356 from hikouki/add-docs-on-responsive-breakpoints

Add docs for responsive breakpoints.

IamManchanda committed:
Added docs on block grid!

IamManchanda committed:
Add codepen example and also move block grid below offset docs!

kball committed:
Merge pull request #10357 from zurb/docs/add-block-grid

Added docs on block grid!

stuartstarrs committed:
Changed file permissions from 755 to 644

kball committed:
Merge pull request #10359 from stuartstarrs/develop

Changed file permissions from 755 to 644

IamManchanda committed:
Add Button cursor property and bind it into a variable
so that user can change it!

IamManchanda committed:
Add a test case for cursor button!

IamManchanda committed:
Fix tabs docs copy!

IamManchanda committed:
Merge pull request #10368 from zurb/docs/fix-tabs-copy

Fix tabs docs copy!

kball committed:
Distributed version of media query should initialize

kball committed:
Merge pull request #10369 from zurb/fix-dist-of-media-query

Distributed version of media query should initialize

kball committed:
Merge pull request #10367 from zurb/default-button-cursor

Add Button cursor property and bind it into a variable

brettsmason committed:
First take on the reworked grid container. Updates docs to show usage as well as modified visual test to show this visually.

Commit:
Option be able to recursively merged.

kball committed:
Merge pull request #10377 from hikouki/extend

Fixed: I made the option posible recursively merge.

Commit:
Set the unposition to use global-left to allow for TRL content to uncenter pushed divs

kball committed:
Complete and essential should now include flex classes

brettsmason committed:
Rework classes to be `.full' and `.fluid`. This required the use of :not so needs testing.

rafibomb committed:
Merge pull request #10406 from zurb/fix-default-customizer-downloads

Complete and essential should now include flex classes

kball committed:
Merge pull request #10405 from Seanom/RTL-centered-push-issue

Resolve a RTL uncentered push/pull issue

IamManchanda committed:
Add spacing b/w else and if!

IamManchanda committed:
Update docs copy of grid container rework!

IamManchanda committed:
Merge pull request #10371 from zurb/grid-container-rework

Grid Container Rework

IamManchanda committed:
Merge pull request #10415 from zurb/hotfix/space-else-if

Add spacing b/w else and if!

SassNinja committed:
loop hasReveal value through to removeContentClasses()

Since the removeContentClasses function gets called within the addContentClasses function,
the hasReveal value must be included to make sure the reveal classes gets updated correctly.

IamManchanda committed:
Add visual test for bug fix `reveal for large` spacing issue!

IamManchanda committed:
Merge pull request #10421 from SassNinja/fix-10407-has-reveal-class

Fix for #10407 reveal content classes

SassNinja committed:
Add visual test for showing any menu in drilldown

SassNinja committed:
Create _showMenu method to open any (sub)menu in a drilldown without clicking through every menu on the way to it.

Compared to the current _show method that targets the submenu's parent, _showMenu targets the submenu itself.
The visual test has got radio buttons now to toggle the methods for an easy comparison.

SassNinja committed:
Add autoHeight option support for _showMenu method

Compared to the current _show method that targets the submenu's parent, _showMenu targets the submenu itself.
The visual test has got radio buttons now to toggle the methods for an easy comparison.

IamManchanda commented:
This is not awesome... this is super awesome !!! Hats off 
Its like deeplinking came into drilldown's :wink: 

cc @kball for merge/review!

IamManchanda reviewed (approved):
:+1:

IamManchanda commented on js/foundation.drilldown.js:
```diff
@@ -343,6 +343,86 @@ class Drilldown extends Plugin {
+    $elem.parent('li').attr('aria-expanded', false);
+    // if (trigger === true) {
+    //   $elem.trigger('hide.zf.drilldown', [$elem]);
+    // }
```
Remove these comments!

SassNinja replied:
The lines are commented in now. Instead of removing the possiblity to trigger a close event I'm  just not using it in _showMenu anymore.

SassNinja committed:
Bring back possibility to trigger event in _setHideSubMenuClasses but remove usage in _showMenu

Instead of commenting the even trigger out it's now commented in but not used anymore by _showMenu because a show action doesn't need to trigger a close event here.

SassNinja committed:
Add param in _showMenu method to disable auto focus of first link in target drilldown (sub)menu

SassNinja commented:
@IamManchanda many thanks for your kindly words! I'm glad to hear this feature is appreciated 😃 

Apart from your requested change (see my comment above) I've made another slight improvement:
The method `_showMenu` has now a second param called `autoFocus` which will prevent the auto focus of the first link in the target (sub)menu if false.
I think this might by helpful in cases where the auto focus is not desired (e.g. if the drilldown is hidden when calling that function). Default remains true.

IamManchanda committed:
Correct Code Indentation!

IamManchanda reviewed (approved):
Seems very Good!

kball commented:
Testing through the visual test I see a few issues

1. When I click in using the 'showMenu' option I get some weird focus behavior (the focus ring which is supposed to only show on keyboard nav starts showing on elements within the drilldown.
2. When I keyboard nav around, I seem to occasionally trigger some sort of behavior where the entire drilldown becomes hidden.

Both of these were in Chrome latest (59) on mac

SassNinja commented:
@kball thanks for your feedback!

1. That's actually a feature: I've added this auto focus so the user can jump right into the drilldown via keyboard after having clicked the trigger (button). Would you prefer no auto focus by default?
Currently the auto focus may get disabled by explicitly setting the param to false
```javascript
$drilldown.foundation('_showMenu', $menu, false);
``` 
2. Would you describe more detailed how to reproduce this?
But I think I know what you mean. When I click 'Show X', then above the buttons (lose focus) and then press several times the TAB key the drilldown menus get mixed up.
However **this isn't caused by my changes / PR** because it also happens if you don't click any button but open "Item A3" the usual way (clicking on it), then click above the buttons (lose focus) and then press several times the TAB key. I think it's a general drilldown focus management issue and may be related to this issue https://github.com/zurb/foundation-sites/issues/10429

SassNinja commented:
ping @kball  – would be great to get your feedback (in particular on the 1st point)

Besides I've noticed this PR and some others are broken, precisely its history.
- https://github.com/zurb/foundation-sites/pull/10404
- https://github.com/zurb/foundation-sites/pull/10456
- https://github.com/zurb/foundation-sites/pull/10394
- https://github.com/zurb/foundation-sites/pull/10408

Surprisingly everathing was fine when I looked into it last time and there haven't been pushes in the meantime (at least in my PRs).
Any idea what might have broken them?

IamManchanda commented:
Poke @kball ... Now what needs to bee done ?

kball commented:
@SassNinja with regards to the PRs getting wonky, unfortunately we had some issues with the develop branch... we've fixed those (and added protection to prevent future occurrences) but old PRs seem to still be a little wonky... going to try closing and reopening to see if that fixes it.

kball commented:
With regards to your questions - I think we can separate #2 into a differnet issue as I've inconsistently seen it in other situations... for #1 we generally don't show the focus ring for non-keyboard nav, so I'd say lets make the default no focus, and then this should be good to go.

Since it doesn't seem that closing and reopening fixed the history, I'm going to close entirely, can you try creating a new PR with these changes and the update above? Thanks!
